grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -540,7 +540,7 @@ class ProductAvailabilityService {
     private static Map<String, Integer> calculateQuantitiesByExpirationFilter(List<ProductAvailability> productAvailabilityRecords, ExpirationFilter expirationFilter) {
         boolean removeExpiredStock = expirationFilter == ExpirationFilter.SUBTRACT_EXPIRED_STOCK
         Date today = new Date()
-        Date maxDate = removeExpiredStock ? today : today + expirationFilter.days
+        Date maxDate = removeExpiredStock ? today : today + (expirationFilter.days ?: 0)
```

Comment:
@ewaterman I take it back, as you caught it in the previous PR. I forgot I moved this part of the code outside the `.sum` closures, and I tested it probably when having it yet inside, hence it was breaking for `DO_NOT_SUBTRACT_EXPIRED_STOCK`.